src/main/java/nl/novi/Les11TechItEasy/Services/TelevisionService.java

```diff
@@ -0,0 +1,174 @@
+        Optional<Television> television = tvRepos.findById(id);
+        if(television.isPresent()){
+            Television t = television.get();
+            tvRepos.delete(t);
+        }else{
+            throw new RecordNotFoundException("Television not found");
+
+        }
+        return null;
+    }
+    public TelevisionOutputDto deleteByName( String name){
+        Optional<Television> television = tvRepos.findByName(name);
+        if(television.isPresent()){
+            Television t = television.get();
+            tvRepos.delete(t);
+        }else
+            throw  new RecordNotFoundException("Television not found");
+        return null;
+    }
+    public  TelevisionDto updateTvList (long id, String name){
+        Optional<Television> television = tvRepos.findById(id);
+        if(television.isPresent()){
+            Television tv = television.get();
+            tv.setName(name);
+            tvRepos.save(tv);
+        }else
+            throw  new RecordNotFoundException("Television not found");
+        return null;
+    }
+
+
+    public TelevisionOutputDto televisionToOutputDto(Television t ){
+        TelevisionOutputDto todto = new TelevisionOutputDto();
+        todto.id = t.getId();
+        todto.brand = t.getBrand();
+        todto.type = t.getType();
+        todto.name = t.getName();
+        todto.price = t.getPrice();
+        todto.availableSize = t.getAvailableSize();
+        todto.refreshRate = t.getRefreshRate();
+        todto.screenType = t.getScreenType();
+        todto.screenQuality = t.getScreenQuality();
+        todto.smartTv = t.isSmartTv();
+        todto.voiceControl = t.isVoiceControl();
+        todto.hdr = t.isHdr();
+        todto.bluetooth = t.isBluetooth();
+        todto.ambiLight = t.isAmbiLight();
+        todto.originalStock = t.getOriginalStock();
+        todto.sold = t.getSold();
+        return todto;
+
+    }
+    public Television DtoToTelevision (TelevisionDto tvDto){
+        Television tv = new Television();
+        tv.setBrand(tvDto.brand);
+        tv.setType(tvDto.type);
+        tv.setName(tvDto.name);
+        tv.setPrice(tvDto.price);
+        tv.setAvailableSize(tvDto.availableSize);
+        tv.setRefreshRate(tvDto.refreshRate);
+        tv.setScreenType(tvDto.screenType);
+        tv.setScreenQuality(tvDto.screenQuality);
+        tv.setSmartTv(tvDto.smartTv);
+        tv.setVoiceControl(tvDto.voiceControl);
+        tv.setHdr(tvDto.hdr);
+        tv.setBluetooth(tvDto.bluetooth);
+        tv.setAmbiLight(tvDto.ambiLight);
+        tv.setOriginalStock(tvDto.originalStock);
+        tv.setSold(tvDto.sold);
+        return tv;
+    }
+
+    public void assingRemoteControllerToTelevision(Long id,Long remoteControllerID){
+        var optionalTelevision = tvRepos.findById(id);
+        var optionalRemteController = rmRepos.findById(remoteControllerID);
+
+        if (optionalTelevision.isPresent()&&optionalRemteController.isPresent()){
```

Comment:
Hoe kom je nu te weten of de television fout is of dat de remotecontroller fout is?